{
  JobQuery query=managementService.createJobQuery().withException();
  verifyQueryResults(query,0);
  ProcessInstance processInstance=runtimeService.startProcessInstanceByKey("exceptionInJobExecution");
  Job timerJob=managementService.createJobQuery().processInstanceId(processInstance.getId()).singleResult();
  assertNotNull("No job found for process instance",timerJob);
  try {
    managementService.executeJob(timerJob.getId());
    fail("RuntimeException from within the script task expected");
  }
 catch (  RuntimeException re) {
    assertTextPresent("This is an exception thrown from scriptTask",re.getMessage());
  }
  query=managementService.createJobQuery().withException();
  verifyQueryResults(query,1);
  timerJob=query.singleResult();
  Assert.assertNotNull(timerJob);
  assertEquals(processInstance.getId(),timerJob.getProcessInstanceId());
  Assert.assertNotNull(timerJob.getExceptionMessage());
  assertTextPresent("This is an exception thrown from scriptTask",timerJob.getExceptionMessage());
}
